### PR TITLE
Bugfix/footer

### LINKED ISSUE #5059: Footer not sticking to the bottom of the screen on pages with little content on large displays
Hi, this is a [first-timers-only issue](https://publiclab.github.io/community-toolbox/#r=all). This means we've worked to make it more legible to folks who either **haven't contributed to our codebase before, or even folks who haven't contributed to open source before**.

If that's you, we're interested in helping you take the first step and can answer questions and help you out as you do. Note that we're especially interested in contributions from people from groups underrepresented in free and open source software!

We know that the process of creating a pull request is the biggest barrier for new contributors. This issue is for you 💝

If you have contributed before, **consider leaving this one for someone new**, and looking through our general [help wanted](https://github.com/publiclab/plots2/labels/help-wanted) issues. Thanks!

### 🤔 What you will need to know.
Nothing. This issue is meant to welcome you to Open Source :) We are happy to walk you through the process.

### 📋 Step by Step
* [ ]  🙋 **Claim this issue**: Comment below. If someone else has claimed it, ask if they've opened a pull request already and if they're stuck -- maybe you can help them solve a problem or move it along!
* [ ]  📝 **Update** edit thw [stylesheets](https://github.com/publiclab/plots2/tree/master/app/assets/stylesheets) folder to add styles that address this issue. You can check out [this article](https://css-tricks.com/couple-takes-sticky-footer/) for help.

**Current Behavior**
Visiting the following pages: _https://publiclab.org/settings_, _https://publiclab.org/subscriptions (when you have little or no subscriptions)_, _https://publiclab.org/subscriptions/digest (when you have little or no subscriptions)_, _https://pubiclab.org/:user/likes (when you have little or no liked content)_ will show a footer that is pushed up and does not stick to the bottom of the page.

![Screenshot 2019-03-14 at 12 23 12 AM](https://user-images.githubusercontent.com/7345686/54321715-85c20d00-45f2-11e9-9140-d9e8bdf46fc7.png)
![Screenshot 2019-03-14 at 12 23 29 AM](https://user-images.githubusercontent.com/7345686/54321722-8e1a4800-45f2-11e9-8c18-9ddd6843b446.png)
![Screenshot 2019-03-14 at 12 23 36 AM](https://user-images.githubusercontent.com/7345686/54321736-9a9ea080-45f2-11e9-9068-122585eeadfb.png)
![Screenshot 2019-03-14 at 12 23 42 AM](https://user-images.githubusercontent.com/7345686/54321744-a1c5ae80-45f2-11e9-9a7b-0a8341f73834.png)

The above images are the only pages with little content I could find on production, but note that the behaviour will be replicated on any page with little content, as evidenced by this screenshot of a page on development:

![Screenshot 2019-03-14 at 12 55 00 AM](https://user-images.githubusercontent.com/7345686/54322055-f3bb0400-45f3-11e9-9389-38020db0761c.png)

                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                          
**Expected Behavior**
The footer should stick to the bottom of the screen, no matter the height of the content.

![Screenshot 2019-03-14 at 1 01 54 AM](https://user-images.githubusercontent.com/7345686/54322375-37623d80-45f5-11e9-8b45-5d6823f07d1a.png)
![Screenshot 2019-03-14 at 1 04 45 AM](https://user-images.githubusercontent.com/7345686/54322376-37623d80-45f5-11e9-84ab-8bdcce4ab754.png)


* [ ]  💾 **Commit** your changes
* [ ]  🔀 **Start a Pull Request**. There are two ways how you can start a pull request:

1. If you are familiar with the terminal or would like to learn it, [here is a great tutorial](https://egghead.io/series/how-to-contribute-to-an-open-source-project-on-github) on how to send a pull request using the terminal.
2. You can also [edit files directly in your browser](https://help.github.com/articles/editing-files-in-your-repository/) and open a pull request from there.

* [ ]  🏁 **Done** Ask in comments for a review :)

### 🤔❓ Questions?
Leave a comment below!

### PR DESCRIPTION
**Pull request by Asu1996**
Fixes #5059  (<=== Add issue number here)

Make sure these boxes are checked before your pull request (PR) is ready to be reviewed and merged. Thanks!

* [x] PR is descriptively titled 📑 and links the original issue above 🔗
* [ ] tests pass -- look for a green checkbox ✔️ a few minutes after opening your PR -- or run tests locally with `rake test`
* [x] code is in uniquely-named feature branch and has no merge conflicts 📁
* [ ] screenshots/GIFs are attached 📎 in case of UI updation
* [ ] ask `@publiclab/reviewers` for help, in a comment below

> We're happy to help you get this ready -- don't be afraid to ask for help, and **don't be discouraged** if your tests fail at first!

If tests do fail, click on the red `X` to learn why by reading the logs.

Please be sure you've reviewed our contribution guidelines at https://publiclab.org/contributing-to-public-lab-software 

Thanks!

### CODE REVIEW

File: app/assets/stylesheets/style.css [hunk 1]
```diff
@@ -528,6 +540,12 @@ a.btn-outline-secondary .fa {
   color: #6c757d;
 }
 
+.container{
```
Comment (jywarren) [inline app/assets/stylesheets/style.css:543]: Hmm, I'm a little worried about modifying default Bootstrap styles here, and about modifying the HTML and body styles... Is there a more minimal way to do this, or is this the only way? 

Would it work to just absolute position the footer? And, I guess, add corresponding amount of bottom-padding to the body so it doesn't overlap anything?

Thanks for working through this with us, just trying to be sure it works on a variety of screen sizes and is maintainable and Bootstrap compatible! Thanks a lot!

**Comment by jywarren** [comment]
What do you think, @avats-dev of my questions above? Thanks both of you!!!

**Comment by avats-dev** [comment]
As far as I know, you will need to modify body container along with footer div for achieving this. Setting it absolute has a drawback that if there is more content, even then footer will be stuck at bottom. So, it's best to attach it to page container instead of viewport.  

You can see [this](https://stackoverflow.com/questions/25671073/footer-below-content-but-not-floating-mid-air-if-not-enough-content) for reference.

We can also use flex, but that will also require editing body div.

**Comment by Asu1996** [comment]
@avats-dev Sir, should i work on doing a better code?

**Comment by jywarren** [comment]
Hi, well, hmm -- in testing this in GitPod, i see some other layout issues resulting - like, here on tag pages: `/tag/test` -- there's extra white space at the top of the page:

image

(compare to https://publiclab.org/tag/outreachy)

However, other pages seem OK! Even in mobile-friendly page-widths:

image

However, I see as @avats-dev [mentioned](https://github.com/publiclab/plots2/issues/5059#issuecomment-706640382), this doesn't happen much because our footer is quite full of content now. 

Balancing all of this, I guess I think we'd better leave the layout more standard and not make these changes, given that any customization will make our layouts slightly more brittle and harder to maintain. I want to apologize to @Asu1996 who put good work into this and solved the stated issue, and will make a new issue for @Asu1996 to make up for this, hopefully. 

Thanks, everyone, for your help on this, and my apologies for the reversal. I appreciate your patience!

**Comment by jywarren** [comment]
I'm pulling one from https://github.com/publiclab/plots2/issues/8545 - will post the issue here for @Asu1996 as soon as it's posted!

**Comment by jywarren** [comment]
Here you go @Asu1996 ! https://github.com/publiclab/plots2/issues/8546

**Pull request closed without merging**